common-files/utils/httputils.mjs

```diff
@@ -234,18 +236,59 @@ const responseLogger = (req, res, next) => {
   }
 };
 
+const isEndpointWhitelisted = req => {
+  let result = false;
+  req.app.get(ENDPOINTS_WHITELISTED).forEach(e => {
+    if (!result) {
+      result = req.url.match(e) != null;
+    }
+  });
+
+  return result;
+};
+
+const authenticationHandler = (req, res, next) => {
+  if (
+    req.url === '/healthcheck' ||
+    isEndpointWhitelisted(req) ||
+    req.get('x-api-key') === process.env.AUTHENTICATION_KEY
+  ) {
+    return next();
+  }
+
+  res.sendStatus(401);
+
+  return null;
+};
+
 /**
- * Setup the default filters for the app being passed as parameter. This can be extended later to allow
- * additional parameters to be passed to customize the filters/handlers.
+ * Setup the default filters for the app being passed as parameter. It also enables endpoint authentication when
+ * the environment variable AUTHENTICATION_KEY is set: this is the key that will be used to authenticate requests
+ * against the request header 'X-API-Key'. Endpoints can be whitelisted by using the environment variable
+ * ENDPOINTS_WHITELISTED listing the whitelisted endpoints (e.g. ENDPOINTS_WHITELISTED="/commitment/save, /commitment/delete").
+ * The env var ENDPOINTS_WHITELISTED accepts regex expression values.
  *
- * @param {*} app - an instance of 'expressjs'.
+ * @param {*} app - an instance of 'expressjs'
+ * @param routesDefiner - an annonymous function that receives an expressjs instance and defines the routes
+ * @param addHealthCheck - adds a /healthcheck endpoint automatically
+ * @param useFileUpload - adds a middleware for file uploading
```

Comment:
Since we are here can we add types to this function? Like so…

```
 * @param {*} app - an instance of 'expressjs'
  * @param {Function} routesDefiner - an annonymous function that receives an expressjs instance and defines the routes
  * @param {boolean} addHealthCheck - adds a /healthcheck endpoint automatically
  * @param {boolean} useFileUpload - adds a middleware for file uploading
```

Reply:
Good one!